### PR TITLE
Fix dequantizelinaer simple case with no zero point present

### PR DESCRIPTION
**Pull request by TedThemistokleous**
Dequantielinear was failing when only two inputs were given (input, scale) as the zero point type check was applied always instead of when there's 3 inputs. Missed due to missing test cases in verify which would have exercised compute_shape() for the op

Reused existing generated onnx files in parser.

Arose when adding changes for #3445  which uses dequantizelinear before the final dot is produced.

### CODE REVIEW

**Review by shivadbhavsar — APPROVED** [review]

File: src/include/migraphx/op/dequantizelinear.hpp [hunk 1]
```diff
@@ -49,10 +49,13 @@ struct dequantizelinear
     std::string name() const { return "dequantizelinear"; }
     shape compute_shape(std::vector<shape> inputs) const
     {
-        check_shapes{inputs, *this}.same_dims().has(2, 3);
-        if(inputs.size() == 3 and inputs[0].type() != inputs[2].type())
+        if(inputs.size() == 3)
```
Comment (lakhinderwalia) [inline src/include/migraphx/op/dequantizelinear.hpp:52]: Thanks for fixing this, but perhaps this Operator is still missing an important check when there are 2 or 3  inputs. And `x_scale` is a 1-D Tensor. We should check this 1-D Tensor against the axis dimension of `X`. I don't see that check in parse_dequantizelinear.cpp either. Is it done somewhere else? Thanks.

**Review by lakhinderwalia — APPROVED** [review]
Approving this. If you want to fix something additional (left a comment) if that is an issue, please feel free to fix it as well.

File: src/include/migraphx/op/dequantizelinear.hpp (hunk 1 above)
Comment (TedThemistokleous) [reply src/include/migraphx/op/dequantizelinear.hpp:52]: I believe it should be. If not I'll make another issue and PR for that case

File: src/include/migraphx/op/dequantizelinear.hpp [hunk 2]
```diff
@@ -49,10 +49,13 @@ struct dequantizelinear
...
         {
-            MIGRAPHX_THROW("DEQUANTIZELINEAR: Zero point and input should be the same type.");
+            check_shapes{inputs, *this}.same_dims().has(2, 3);
```
Comment (pfultz2) [inline src/include/migraphx/op/dequantizelinear.hpp:54]: Why is this moved down here? We should still check that the inputs are the same dims even when 2 inputs are given. It makes sense up at the top.

File: src/include/migraphx/op/dequantizelinear.hpp (hunk 1 above)
Comment (pfultz2) [reply src/include/migraphx/op/dequantizelinear.hpp:52]: > And x_scale is a 1-D Tensor. We should check this 1-D Tensor against the axis dimension of X.

No, this is not the onnx `DequantizeLinear` operator. In migraphx, all inputs should have the same dimensions(which is what `check_shapes` checks for). If in onnx, the `x_scale` is a 1-D tensor then a broadcast is inserted to make it have the same dimensions here:

https://github.com/ROCm/AMDMIGraphX/blob/fab88b0eca40b85d8bd53a40ff75acc587ed019d/src/onnx/quantize_dequantize_linear.cpp#L70

**Review by pfultz2 — CHANGES_REQUESTED** [review]
We should still check that all inputs have the same dimensions even when 2 parameters are used.

File: src/include/migraphx/op/dequantizelinear.hpp (hunk 1 above)
Comment (lakhinderwalia) [reply src/include/migraphx/op/dequantizelinear.hpp:52]: Thanks, Paul. For the onnx case, I do see the 1D tensor shape is being checked here: [quantize_dequantize_linear.cpp](https://github.com/ROCm/AMDMIGraphX/blob/fab88b0eca40b85d8bd53a40ff75acc587ed019d/src/onnx/quantize_dequantize_linear.cpp#L60)

File: src/include/migraphx/op/dequantizelinear.hpp (hunk 2 above)
Comment (TedThemistokleous) [reply src/include/migraphx/op/dequantizelinear.hpp:54]: That wont be the case though in this operator would it? I guess we broadcast out scale if its a scalar or its 1-d for the case of scale being a tensor

Comment (TedThemistokleous) [reply src/include/migraphx/op/dequantizelinear.hpp:54]: That's where the issue was coming from when invoking dequantize_linear, zero points weren't present and resulted in an error.

Comment (pfultz2) [reply src/include/migraphx/op/dequantizelinear.hpp:54]: What was the error specifically? This allows either 2 or 3 inputs so it shouldn't throw an error for 2 inputs.

Comment (TedThemistokleous) [reply src/include/migraphx/op/dequantizelinear.hpp:54]: image

would invoke

```[   RUN    ] matmulintegertofloat_test
terminate called after throwing an instance of 'migraphx::version_1::exception'
  what():  /code/AMDMIGraphX/src/include/migraphx/check_shapes.hpp:220: same_dims: dequantizelinear: Dimensions do not match```

Comment (TedThemistokleous) [reply src/include/migraphx/op/dequantizelinear.hpp:54]: Unless you want us to always broadcast the scale dimensions (even if its 1-d) but then that means we're inserting a broadcast to remove later. For the scale here, we're never supposed to have the same dimensions between input and scale vectors. Its why I gated this check by input size.

**Comment by TedThemistokleous** [comment]
Not a bug. In MIGraphX we handle this as pointwise so we need to have everything broadcasted to the input to avoid index operations
